logdna/resource_stream_exclusion.go

```diff
@@ -0,0 +1,185 @@
+package logdna
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type streamExclusion struct {
+	ID     string   `json:"id,omitempty"`
+	Title  string   `json:"title"`
+	Active bool     `json:"active"`
+	Apps   []string `json:"apps"`
+	Hosts  []string `json:"hosts"`
+	Query  string   `json:"query"`
+}
+
+var streamExclusionAtLeastOneOfFields = []string{"apps", "hosts", "query"}
+
+func resourceStreamExclusionCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	pc := m.(*providerConfig)
+	ex := streamExclusion{
```

Comment:
Nit: for consistency's sake and readability, might be a good idea to rename this variable to `exclusion`

Reply:
this is more of a style thing common to a lot of golang - I tend to fall back to this document for style things (for anything not enforced by the static analysis tooling in `golang-lint`)

https://github.com/golang/go/wiki/CodeReviewComments#variable-names